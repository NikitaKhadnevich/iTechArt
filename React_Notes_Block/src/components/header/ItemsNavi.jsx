/* eslint-disable no-unused-vars */
import React from 'react';
import { useSelector } from 'react-redux';
import { Typography } from '@material-ui/core';
import {
  notes,
  sharedNotes,
  about,
  signUp,
  signIn,
  isSignIn,
  userData,
} from './headerReceiver';

import { Navigation, NavLink, SignBox } from './styled';

const ItemsNavi = () => {
  const userIsSignIn = useSelector(isSignIn);
  const sharedNotesLength = useSelector(userData);
  const SHARED_COUNT = 0;

  return (
    <Navigation>
      {userIsSignIn && (
        <>
          <NavLink to={notes} data-path={notes}>
            <Typography variant='h6'>Notes</Typography>
          </NavLink>
          <NavLink to={sharedNotes} data-path={sharedNotes}>
            <Typography variant='h6'>
              Shared Notes (
              {sharedNotesLength[0]?.sharedNotes
                ? sharedNotesLength[0]?.sharedNotes.length
                : SHARED_COUNT}
              )
            </Typography>
          </NavLink>
        </>
      )}
      <NavLink to={about} data-path={about}>
        <Typography variant='h6'>About</Typography>
      </NavLink>
      {!userIsSignIn && (
        <SignBox>
          <NavLink to={signUp} data-path={signUp}>
            <Typography variant='body2'>Sign Up</Typography>
          </NavLink>
          <NavLink to={signIn} data-path={signIn}>
            <Typography variant='body2'>Sign In</Typography>
          </NavLink>
        </SignBox>
      )}
    </Navigation>
  );
};
export default ItemsNavi;
