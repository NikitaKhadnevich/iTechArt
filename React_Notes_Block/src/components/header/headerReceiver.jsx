/* eslint-disable no-unused-vars */
import {
  isSignIn,
  userData,
  isSignUp,
  isSignUpError,
  isSignInError,
} from '../../api/signToolkit/signSelectors';
import ROUTERS from '../../config/routers/routers';
import {
  SET_AUTH_USER_DATA,
  IS_SIGN_IN,
  IS_SIGN_UP,
  SET_SIGNUP_ERROR,
  SET_SIGNIN_ERROR,
  GET_SIGN_OUT,
} from '../../api/signToolkit/signToolkit';

import {
  setToLocalStorage,
  getToLocalStorage,
} from '../../utils/localStorage/SetGetLocStor';

import {
  AUTH_LOCAL_DATA,
  AUTH_LOCAL_STATUS,
} from '../../config/constants/localStoreKeys';

import {
  SET_NOTES_LIST,
  SET_USERS_LIST,
  SET_INFINITE_FETCH_COUNTER,
} from '../../api/notesToolkit/notesToolkit';

import { InfiniteNotes } from '../../config/constants/queryKeys';

const { notes, notFound, sharedNotes, about, signUp, signIn } = ROUTERS;

export {
  notes,
  notFound,
  sharedNotes,
  about,
  signUp,
  signIn,
  isSignIn,
  userData,
  SET_AUTH_USER_DATA,
  IS_SIGN_IN,
  IS_SIGN_UP,
  SET_SIGNUP_ERROR,
  SET_SIGNIN_ERROR,
  GET_SIGN_OUT,
  SET_NOTES_LIST,
  SET_USERS_LIST,
  SET_INFINITE_FETCH_COUNTER,
  isSignInError,
  setToLocalStorage,
  getToLocalStorage,
  AUTH_LOCAL_DATA,
  AUTH_LOCAL_STATUS,
  InfiniteNotes,
};
