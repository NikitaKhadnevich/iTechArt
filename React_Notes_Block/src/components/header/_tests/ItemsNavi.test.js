/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-unused-vars */
import React from 'react';
import * as reactRedux from 'react-redux';
import { Provider } from 'react-redux';
import { configure, shallow, mount, create, render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import configureMockStore from 'redux-mock-store'; // ES6 modules
import { BrowserRouter as Router } from 'react-router-dom';
import ItemsNavi from '../ItemsNavi';
import { isSignIn, sharedNotes, userData } from '../headerReceiver';

configure({ adapter: new Adapter() });

describe('Header', () => {
  const middlewares = [];
  const mockStore = configureMockStore(middlewares);
  const initialState = { status: true }; // Init StorContain
  const store = mockStore(initialState);
  const { status } = store.getState(); // useSelector

  beforeEach(() => {
    const useSelector = jest.spyOn(reactRedux, 'useSelector');
    useSelector.mockReturnValue(status);
    if (status) {
      useSelector.mockReturnValue([{ sharedNotes: [1, 2, 3, 4, 5] }]);
    }
    const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  let ItemsWrapper;
  beforeEach(() => {
    ItemsWrapper = shallow(
      <Provider store={store}>
        <Router>
          <ItemsNavi />
        </Router>
      </Provider>
    );
  });

  it('Is it ItemsNavi Exist', () => {
    expect(ItemsWrapper).toHaveLength(1);
  });
  it('Is it ItemsNavi User Sign In And has Shared Notes', () => {
    const ItemsWrapper = shallow(<ItemsNavi />);
    expect(ItemsWrapper).toMatchSnapshot();
  });
  it('should contain Links', () => {
    const ItemsWrapper = shallow(<ItemsNavi />);
    expect(
      ItemsWrapper.find('WithStyles(ForwardRef(Typography))')
    ).toHaveLength(3);
  });
  it('should contain Notes or About Nav', () => {
    const ItemsWrapper = shallow(<ItemsNavi />);
    expect(
      ItemsWrapper.find('WithStyles(ForwardRef(Typography))').first().text()
    ).toEqual('Notes');
  });
  it('should contain Notes', () => {
    expect(
      ItemsWrapper.findWhere(
        (n) =>
          n.type() === 'WithStyles(ForwardRef(Typography))' &&
          n.contains('Notes')
      )
    );
  });
  it('should contain Sign Up', () => {
    expect(
      ItemsWrapper.findWhere(
        (n) =>
          n.type() === 'WithStyles(ForwardRef(Typography))' &&
          n.contains('Sign Up')
      )
    );
  });
  it('should contain Sign In', () => {
    expect(
      ItemsWrapper.findWhere(
        (n) =>
          n.type() === 'WithStyles(ForwardRef(Typography))' &&
          n.contains('Sign In')
      )
    );
  });
  it('should contain Shared Notes with Shared Data', () => {
    expect(
      ItemsWrapper.findWhere(
        (n) =>
          n.type() === 'WithStyles(ForwardRef(Typography))' &&
          n.contains(`Shared Notes (5ee)`)
      )
    );
  });
});
