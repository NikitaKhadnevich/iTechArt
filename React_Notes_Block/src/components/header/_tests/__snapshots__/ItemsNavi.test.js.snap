// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Is it ItemsNavi User Sign In And has Shared Notes 1`] = `
<Styled(Styled(MuiBox))>
  <Styled(Link)
    data-path="/notes"
    to="/notes"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      Notes
    </WithStyles(ForwardRef(Typography))>
  </Styled(Link)>
  <Styled(Link)
    data-path="/shared-notes"
    to="/shared-notes"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      Shared Notes (
      5
      )
    </WithStyles(ForwardRef(Typography))>
  </Styled(Link)>
  <Styled(Link)
    data-path="/about"
    to="/about"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      About
    </WithStyles(ForwardRef(Typography))>
  </Styled(Link)>
</Styled(Styled(MuiBox))>
`;
